src/marqo/_httprequests.py

```diff
@@ -32,6 +32,13 @@ def _operation(self, method: HTTP_OPERATIONS) -> Callable:
 
         return OPERATION_MAPPING[method]
 
+    def construct_path(self, path: str) -> str:
+        """Augment the URL request path based if telemetry is required."""
+        if self.config.use_telemetry:
+            delimeter= "?" if "?" not in f"{self.config.url}/{path}" else "&"
```

Comment:
I suggest `not path.endswith('?')` instead of `"?" not in f"{self.config.url}/{path}"` as the former is more specific.

Reply:
I don't think this would work. Consider `http://localhost:8882/indexes/bulk/search?query=param`. We'd get
```python
self.config.url="http://localhost:8882"
path="indexes/bulk/search?query=param"
```
Then `("?" if  not path.endswith('?') else "&") == "?"` not `"&"`

Reply:
You are right!